From 34e1c047d2a80772e22816c36d41822ebdee9666 Mon Sep 17 00:00:00 2001
From: Manigandan Gopalakrishnan <manigandan.gopalakrishnan@ltts.com>
Date: Tue, 13 Sep 2022 22:31:53 +0530
Subject: [PATCH] REFPLTB-1808: Observing CcspWifiagent crash in RPI4 64-bit build

Reason for change: Checking 0 for strLoc to avoid SIGSEGV crash
Test procedure: Found that wifi-agent is not crashing
Risks: None

Signed-off-by: Manigandan Gopalakrishnan <manigandan.gopalakrishnan@ltts.com>
Change-Id: I717a1104e45f8d5ecd329e66a074f0a17d4a4c97
---

diff --git a/source/TR-181/sbapi/cosa_wifi_apis.c b/source/TR-181/sbapi/cosa_wifi_apis.c
index a8eb06c..a748bbb 100644
--- a/source/TR-181/sbapi/cosa_wifi_apis.c
+++ b/source/TR-181/sbapi/cosa_wifi_apis.c
@@ -11858,7 +11858,10 @@
         strCount = rc;
         strLoc += strCount;
     }
-    pInfo->PossibleChannels[strLoc-1] = '\0';
+    if(strLoc) //To avoid SIGSEGV when strLoc is 0.
+    {
+        pInfo->PossibleChannels[strLoc-1] = '\0';
+    }
     ccspWifiDbgPrint(CCSP_WIFI_TRACE, "\n %s PossibleChannels : %s\n", __FUNCTION__, pInfo->PossibleChannels);

     /*Update AutoChannelSupported per radio*/
@@ -11878,7 +11881,10 @@
         strCount = rc;
         strLoc += strCount;
     }
-    pInfo->TransmitPowerSupported[strLoc-1] = '\0';
+    if(strLoc) //To avoid SIGSEGV when strLoc is 0.
+    {
+        pInfo->TransmitPowerSupported[strLoc-1] = '\0';
+    }
     ccspWifiDbgPrint(CCSP_WIFI_TRACE, "%s TransmitPowerSupported : %s\n", __FUNCTION__, pInfo->TransmitPowerSupported);

     pInfo->CipherSupported = wifiRadioCap->cipherSupported;
